delayed.go

```diff
@@ -0,0 +1,78 @@
+package transport
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+type DelayedConfig struct {
+	// RequestDelayMin is the delay before the request is sent.
+	RequestDelayMin time.Duration
+
+	// RequestDelayMax is the maximum delay before the request is sent.
+	RequestDelayMax time.Duration
+
+	// ResponseDelayMin is the delay before the response is returned.
+	ResponseDelayMin time.Duration
+
+	// ResponseDelayMax is the maximum delay before the response is returned.
+	ResponseDelayMax time.Duration
+}
```

Comment:
Do you expect anyone to set up different request / response delay in their tests?

I wonder if we could simplify the API a bit:

`transport.Delayed(min, max time.Duration)`

Or perhaps have two transports:
```go
transport.Chain(
    http.DefaultTransport,
    transport.DelayedRequest(min, max),
    transport.DelayedResponse(min, max),
}
```

Or do you feel we'll be adding more options to this transport over time?

Reply:
Sounds good